ConfigurationSystem/Agent/RucioSynchronizer.py

```diff
@@ -0,0 +1,359 @@
+                    else:
+                        if key == 'SpaceToken':
+                            space_token = value
+                        if params['scheme'] == 'srm' and key == 'WSUrl':
+                            params['extended_attributes'] = {'web_service_path': '%s' % value, 'space_token': space_token}
+                    if key == 'Protocol':
+                        params['domains']['lan']['read'] = read_protocols.get(value, 0)
+                        params['domains']['wan']['read'] = read_protocols.get(value, 0)
+                        params['domains']['lan']['write'] = write_protocols.get(value, 0)
+                        params['domains']['wan']['write'] = write_protocols.get(value, 0)
+                        params['domains']['lan']['delete'] = write_protocols.get(value, 0)
+                        params['domains']['wan']['delete'] = write_protocols.get(value, 0)
+                        params['domains']['wan']['third_party_copy'] = write_protocols.get(value, 0)
+                seProtocols[seName].append(params)
+    return S_OK(seProtocols)
+
+
+class RucioSynchronizer(AgentModule):
+    """
+    .. class::  RucioSynchronizer
+
+    Agent that synchronizes Rucio and Dirac
+    """
+
+    def initialize(self):
+        """ agent's initialisation
+
+        :param self: self reference
+        """
+        self.log =  gLogger.getSubLogger('RucioSynchronizer')
+        self.log.info("Starting RucioSynchronizer")
+        return S_OK()
+
+    def execute(self):
+        """ execution in one agent's cycle
+
+        :param self: self reference
+        """
+
+        valid_protocols = ['srm', 'gsiftp', 'davs', 'https', 'root']
+        default_email = None
+        try:
+            client = Client(account='root', auth_type='userpass')
+
+            # Get the storage elements from Dirac Configuration and create them in Rucio
+            newRSE = False
+            self.log.info("Synchronizing SEs")
+            result = getStorageElements()
+            if result['OK']:
+                rses = [rse['rse'] for rse in client.list_rses()]
+                for se in result['Value']:
+                    if se not in rses:
+                        # The SE doesn't exist. Will create it
+                        newRSE = True
+                        self.log.info('%s does not exist and will be created' % se)
+                        try:
+                            client.add_rse(rse=se, deterministic=True, volatile=False)
+                        except Exception as err:
+                            self.log.error('Cannot create RSE' % str(err))
+                            continue
+
+                        # Add RSE attributes for the new RSE
+                        dictRSEAttributes = {'naming_convention': 'BelleII', 'ANY': True, 'fts': 'https://fts.usatlas.bnl.gov:8446'}
+                        dictRSEAttributes = {'naming_convention': 'BelleII', 'ANY': True, 'fts': 'https://kek2-fts01.cc.kek.jp:8446'}
```

Comment:
I guess this should come from the CS?

Reply:
Yes. Any suggestion in which section ?

Reply:
The list of FTS servers is `/Resources/FTSEndpoints`